src/test/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/ReceivedMessagesHealthControllerTest.java

```diff
@@ -191,6 +197,48 @@ void test_health_reports_success_if_messages_check_disabled_in_current_environme
         verifyNoInteractions(caseEventMessageRepository);
     }
 
+    @ParameterizedTest
+    @MethodSource(value = "enabledEnvProvider")
+    void test_enabled_for_environment_return_false_prod_aat(String env) {
+
+        setField(receivedMessagesHealthController, "receivedMessageCheckEnvEnabled", "prod,aat");
+
+        MockHttpServletRequest mockHttpServletRequest = new MockHttpServletRequest();
+        mockHttpServletRequest.setServerName("case-event-handler.any.environment");
+        mockHttpServletRequest.setRequestURI("/health");
+
+        RequestContextHolder.setRequestAttributes(new ServletRequestAttributes(mockHttpServletRequest));
+
+        assertFalse(receivedMessagesHealthController.isNotEnabledForEnvironment(env));
+    }
+
+    @ParameterizedTest
+    @MethodSource(value = "disabledEnvProvider")
+    void test_enabled_for_environment_return_true_demo_ithc(String env) {
+
+        setField(receivedMessagesHealthController, "receivedMessageCheckEnvEnabled", "prod,aat");
+
+        MockHttpServletRequest mockHttpServletRequest = new MockHttpServletRequest();
+        mockHttpServletRequest.setServerName("case-event-handler.any.environment");
+        mockHttpServletRequest.setRequestURI("/health");
+
+        RequestContextHolder.setRequestAttributes(new ServletRequestAttributes(mockHttpServletRequest));
+
+        assertTrue(receivedMessagesHealthController.isNotEnabledForEnvironment(env));
+    }
+
+    @Test
+    void test_enabled_for_environment_return_true_staging_aat() {
+
+        MockHttpServletRequest mockHttpServletRequest = new MockHttpServletRequest();
+        mockHttpServletRequest.setServerName("case-event-handler.staging.aat");
+        mockHttpServletRequest.setRequestURI("/health");
+
+        RequestContextHolder.setRequestAttributes(new ServletRequestAttributes(mockHttpServletRequest));
+
+        assertTrue(receivedMessagesHealthController.isNotEnabledForEnvironment("aat"));
+    }
+
```

Comment:
A general comment since we are using the Http Request Context from Spring we should have a Spring Integration test that loads the Spring context.  That way you will know that the host name is present etc within the context.  

Here we are obviously just mocking all this.